src/main/kotlin/me/steven/indrev/blockentities/farms/FarmerBlockEntity.kt

```diff
@@ -102,8 +101,21 @@ class FarmerBlockEntity(tier: Tier) : AOEMachineBlockEntity<BasicMachineConfig>(
                     droppedStacks.forEach { inventory.output(it) }
                     true
                 }
-                block == Blocks.AIR && canPlant(item) && stack.count > 1 -> {
-                    val cropState = (item as BlockItem).block.defaultState
+                block is AirBlock && canPlant(item) && stack.count > 1 -> {
+                    var cropState = (item as BlockItem).block.defaultState
+
+                    if (item.block is CocoaBlock) {
+                        val directions = arrayOf(Direction.NORTH, Direction.EAST, Direction.SOUTH, Direction.WEST)
+
+                        directions.forEach loop@ {
+                            var supportBlock = world.getBlockState(pos.offset(it)).block
+                            if (supportBlock == Blocks.JUNGLE_LOG || supportBlock == Blocks.JUNGLE_WOOD || supportBlock == Blocks.STRIPPED_JUNGLE_LOG || supportBlock == Blocks.STRIPPED_JUNGLE_WOOD) {
+                                cropState = cropState.with(CocoaBlock.FACING, it)
+                                return@loop
```

Comment:
This should be replaced with a normal for loop because return here doesn't break the forEach, it just moves on the next iteration